File: Packets/HandshakeChallenge.vb
Public Class HandshakeChallenge
    Inherits Packet
    Public Overrides Property OPCode As Byte = &H3
    Public Property Salt As String
    Public Property Rounds As UInteger
    Sub New(ByVal _Salt As String, Optional ByVal _Rounds As Integer = 5000)
        MyBase.New()
        Salt = _Salt
        Rounds = _Rounds
    End Sub
    Public Overrides Function GetByteArray() As Byte()
        Dim p As New PacketBuilder
        p.Write("")
        p.Write(Salt)
        p.Write(Rounds)
        Payload = p.GetBytes
        Return Package(False, True)
    End Function
End Class

File: Packets/HandshakeChallenge.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
public class HandshakeChallenge : Packet
{
	public override byte OPCode { get; set; }
	public string Salt { get; set; }
	public int Rounds { get; set; }
	public HandshakeChallenge(string _Salt, int _Rounds = 5000) : base()
	{
        OPCode = 0x3;
		Salt = _Salt;
        Rounds = _Rounds;
	}
	public override byte[] GetByteArray()
	{
		PacketBuilder p = new PacketBuilder();
		p.Write(((byte)0));
		p.Write(Salt);
		p.Write(Rounds);
		Payload = p.GetBytes();
        byte[] lel = Package();
        return Package();
	}
}
